src/sentry/dynamic_sampling/sliding_window_rebalancing.py

```diff
@@ -0,0 +1,123 @@
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry import options
+from sentry.dynamic_sampling.rules.utils import OrganizationId, ProjectId
+from sentry.dynamic_sampling.snuba_utils import MAX_TRANSACTIONS_PER_PROJECT
+from sentry.sentry_metrics import indexer
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.metrics.naming_layer.mri import TransactionMRI
+from sentry.snuba.referrer import Referrer
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+MAX_SECONDS = 60
+CHUNK_SIZE = 9998  # Snuba's limit is 10000, and we fetch CHUNK_SIZE+1
+QUERY_TIME_INTERVAL = timedelta(
+    hours=24
+)  # By default, we want to get the volume of the last 24 hours.
+
+
+def fetch_projects_with_total_root_transactions_count(
+    org_ids: List[int],
+    granularity: Optional[Granularity] = None,
+    query_interval: Optional[timedelta] = None,
+) -> Mapping[OrganizationId, Sequence[Tuple[ProjectId, int]]]:
+    """
+    Fetches tuples of (org_id, project_id) and the respective root transaction counts.
+    """
+    if query_interval is None:
+        query_interval = QUERY_TIME_INTERVAL
+        granularity = Granularity(3600)
+
+    count_per_root_metric_id = indexer.resolve_shared_org(
+        str(TransactionMRI.COUNT_PER_ROOT_PROJECT.value)
+    )
+    where = [
+        Condition(Column("timestamp"), Op.GTE, datetime.utcnow() - query_interval),
+        Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+        Condition(Column("metric_id"), Op.EQ, count_per_root_metric_id),
+        Condition(Column("org_id"), Op.IN, list(org_ids)),
+    ]
+
+    # Since we want a uniform distribution of samples, we use the modulo operator and sample with a given sample rate
+    # the orgs.
+    sample_rate = int(options.get("dynamic-sampling.sliding_window_rebalancing.sample_rate") * 100)
+    if sample_rate != 100:
+        where += [Condition(Function("modulo", [Column("org_id"), 100]), Op.LT, sample_rate)]
+
+    start_time = time.time()
+    offset = 0
+    aggregated_projects = defaultdict(list)
+    while (time.time() - start_time) < MAX_SECONDS:
+        query = (
+            Query(
+                match=Entity(EntityKey.GenericOrgMetricsCounters.value),
+                select=[
+                    Function("sum", [Column("value")], "root_count_value"),
+                    Column("org_id"),
+                    Column("project_id"),
+                ],
+                where=where,
+                groupby=[Column("org_id"), Column("project_id")],
+                orderby=[
+                    OrderBy(Column("org_id"), Direction.ASC),
+                    OrderBy(Column("project_id"), Direction.ASC),
+                ],
+                granularity=granularity,
+            )
+            .set_limitby(
+                LimitBy(
+                    columns=[Column("org_id"), Column("project_id")],
+                    count=MAX_TRANSACTIONS_PER_PROJECT,
+                )
+            )
```

Comment:
No need for this, you can safely remove it.

This clause made sense for retrieving transactions when we wanted to limit the number of records (i.e. transactions) returned for each (project,org).

There are no transactions here so you will have only one aggregation record for each ( proj, org) tuple.